packages/swift/Sources/Generated.swift

```diff
@@ -235,7 +235,26 @@ class WidgetEventDispatcher: Dispatcher {
         self.delegate = delegate
     }
 
-    func dispatch(_ url: URL) {}
+    func dispatch(_ url: URL) {
+        guard let event = parse(url) else {
+            return
+        }
+
+        delegate.widgetEvent(event)
+
+        switch event {
+        case let event as WidgetEvent.Load:
+            delegate.widgetEvent(event)
```

Comment:
`event` is of type `WidgetEvent.Load` here, so Swift correctly calls the `widgetEvent(_: WidgetEvent.Load)` method. This is how we can keep calling `delegate.widgetEvent` throughout this file. As long as we're passing an appropriately typed value Swift will call the expected method.

Reply:
I know this generated and all so maybe it just doesn't matter, but would it be better/easier/possible to group the cases?

Reply:
Not sure I follow. What do you mean by group the cases?

Reply:
Like keep all the events grouped in the code by their respective types? They already are. [`WidgetEvent` cases](https://github.com/mxenabled/widget-post-message-definitions/pull/33/files#diff-31963ceb4816145a375f0276158a39a436799b3f3eb932d72eaeba4f16bbb0e8R294-R301) are followed by [`ConnectWidgetEvent` cases](https://github.com/mxenabled/widget-post-message-definitions/pull/33/files#diff-31963ceb4816145a375f0276158a39a436799b3f3eb932d72eaeba4f16bbb0e8R302-R329).

Reply:
I'm really just talking about nit pick syntax:

Something like:
```
        case let event as WidgetEvent.Load:
        case let event as WidgetEvent.Ping:
        case let event as WidgetEvent.Navigation:
            delegat.widgetEvent(event)
```

Just so it isn't calling the same thing for each branch? Again, this isn't a big deal.

Reply:
Oh gotcha. No, we can't do that because Swift wouldn't know what the type of `event` is (is it a `WidgetEvent.Load`, or a `WidgetEvent.Ping`, or a `WidgetEvent.Navigation`, etc.) and therefore wouldn't be able to dispatch to the right `widgetEvent` method.

Reply:
I wish we could, because then that method would just be

```swift
// ...

delegate.widgetEvent(event as! Event)
delegate.widgetEvent(event)
```

Or something like that.